modules/lang-painless/src/main/java/org/elasticsearch/painless/symbol/IRDecorations.java

```diff
@@ -74,15 +84,328 @@ public String toString() {
 
     public static class IRDFlags extends IRDecoration<Integer> {
 
-        public IRDFlags(Integer flags) {
-            super(Objects.requireNonNull(flags));
+        public IRDFlags(Integer value) {
+            super(value);
+        }
+    }
+
+    public static class IRCAllEscape implements IRCondition {
+
+        private IRCAllEscape() {
+
+        }
+    }
+
+    public static class IRDCast extends IRDecoration<PainlessCast> {
+
+        public IRDCast(PainlessCast value) {
+            super(value);
+        }
+    }
+
+    public static class IRDExceptionType extends IRDType {
+
+        public IRDExceptionType(Class<?> value) {
+            super(value);
+        }
+    }
+
+    public static class IRDSymbol extends IRDecoration<String> {
+
+        public IRDSymbol(String value) {
+            super(value);
+        }
+    }
+
+    public static class IRDComparisonType extends IRDType {
+
+        public IRDComparisonType(Class<?> value) {
+            super(value);
+        }
+    }
+
+    public static class IRDConstant extends IRDecoration<Object> {
+
+        public IRDConstant(Object value) {
+            super(value);
+        }
+    }
+
+    public static class IRDDeclarationType extends IRDType {
+
+        public IRDDeclarationType(Class<?> value) {
+            super(value);
+        }
+    }
+
+    public static class IRDName extends IRDecoration<String> {
+
+        public IRDName(String value) {
+            super(value);
+        }
+    }
+
+    public static class IRDEncoding extends IRDecoration<String> {
+
+        public IRDEncoding(String value) {
```

Comment:
does this replace `setDefReferenceEncoding`?  If so add the `Def` prefix in there, `Encoding` is too general.

Reply:
Done.